minimesos/src/main/java/com/containersol/minimesos/cluster/ClusterMember.java

```diff
@@ -0,0 +1,42 @@
+package com.containersol.minimesos.cluster;
+
+/**
+ * Generic functionality of every cluster member
+ */
+public interface ClusterMember {
+
+    MesosCluster getCluster();
+    void setCluster(MesosCluster mesosCluster);
+
+    /**
+     * @return the IP address of the container
+     */
+    String getIpAddress();
+
+    /**
+     * Builds container name following the naming convention
+     *
+     * @return container name
+     */
+    String getName();
+
+    /**
+     * @return the ID of the container.
+     */
+    String getContainerId();
+
+    /**
+     * Starts the container and waits until is started
+     *
+     * @param timeout in seconds
+     */
+    void start(int timeout);
+
+    String getRole();
+
+    /**
+     * Removes a container with force
+     */
+    void remove();
+
+}
```

Comment:
Eventually I think we should refactor this to `ClusterProcess` and change container references to process references so it is clear this is a process that can be started, stopped and removed. This also shows that it can have different implementations, either container based or not. We can do it here or in a separate PR.